tests/purs/passing/DerivingFoldable.purs

```diff
@@ -0,0 +1,84 @@
+module Main where
+
+import Prelude
+import Effect.Console (log)
+import Data.Foldable (class Foldable, foldl, foldr, foldMap)
+import Test.Assert
+
+type RecordFields f a =
+  { a :: a
+  -- ensure original field order (not order based on label)
+  -- is order used in fold
+  , zArrayA :: Array a
+  , fa :: f a
+  , ignore :: Int
+  , arrayIgnore :: Array Int
+  , fIgnore :: f Int
+  }
+
+data M f a
+  = M0
+  | M1 a (Array a)
+  | M2 Int
+  | M3 (f a)
+  | M4 (RecordFields f a)
+  | M5 { nested :: RecordFields f a }
+  | M6 Int a (Array Int) (Array a) (f a) (f Int) (RecordFields f a) { nested :: RecordFields f a }
+
+derive instance foldableM :: Foldable f => Foldable (M f)
+
+type MArrStr = M Array String
+
+foldlStr :: forall f. Foldable f => f String -> String
+foldlStr = foldl (<>) ""
+
+foldrStr :: forall f. Foldable f => f String -> String
+foldrStr = foldr (<>) ""
```

Comment:
Happened to notice this in passing; we should test `foldl` and `foldr` with an operation that is expressly *not* monoidal—something like `\l r -> l <> r <> "."` would do—to verify that they fold in the expected directions! (With that example, `foldl` should produce results like `"ab.c."` and `foldr` results like `"abc.."`.)